src/imap.rs

```diff
@@ -542,45 +557,58 @@ impl Imap {
             return 0;
         }
 
-        match self.session.lock().unwrap().0 {
+        let teardown: bool;
+
+        match &mut *self.session.lock().unwrap() {
             Some(ref mut session) => {
                 if let Ok(caps) = session.capabilities() {
-                    let can_idle = caps.has("IDLE");
-                    let has_xlist = caps.has("XLIST");
-
-                    let caps_list = caps.iter().fold(String::new(), |mut s, c| {
-                        s += " ";
-                        s += c;
-                        s
-                    });
+                    if !context.sql.is_open() {
+                        warn!(
+                            context,
+                            0,
+                            "IMAP-LOGIN as {} ok but ABORTING",
+                            as_str(lp.mail_user),
+                        );
+                        teardown = true;
```

Comment:
i added this because if sql is closed already (we are terminating) it doesn't make sense to try to continue with an imap connection.